bin/run.sh

```diff
@@ -22,39 +22,64 @@ if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]; then
-test_output=$(false)
-# TODO: substitute "false" with the actual command to run the test:
-# test_output=$(command_to_run_tests 2>&1)
-
-# Write the results.json file based on the exit code of the command that was 
-# just executed that tested the implementation file
-if [ $? -eq 0 ]; then
-    jq -n '{version: 1, status: "pass"}' > ${results_file}
+# Run the tests for the provided implementation file
+cd "$solution_dir" || exit
+# Output is saved at this location
+echo "==== ${slug}: testing..."
+pwd
+test_output=""
+if ! [ -s *.bal ]; then
+    test_output="WARNING: student did not add code to the balllerina file."
+elif [ ! -e Ballerina.toml ]; then
+    test_output="$test_output \n WARNING: student did not upload Ballerina.toml."
+elif [ ! -e Dependencies.toml ]; then
+    test_output="$test_output \n WARNING: student did not upload Dependencies.toml."
 else
-    # OPTIONAL: Sanitize the output
-    # In some cases, the test output might be overly verbose, in which case stripping
-    # the unneeded information can be very helpful to the student
-    # sanitized_test_output=$(printf "${test_output}" | sed -n '/Test results:/,$p')
-
-    # OPTIONAL: Manually add colors to the output to help scanning the output for errors
-    # If the test output does not contain colors to help identify failing (or passing)
-    # tests, it can be helpful to manually add colors to the output
-    # colorized_test_output=$(echo "${test_output}" \
-    #      | GREP_COLOR='01;31' grep --color=always -E -e '^(ERROR:.*|.*failed)$|$' \
-    #      | GREP_COLOR='01;32' grep --color=always -E -e '^.*passed$|$')
+    # The `--code-coverage` flag generates a test_results.json file
+    test_output=$(bal test --code-coverage)
+fi
+echo $test_output
+
+test_output_file="$solution_dir/target/report/test_results.json"
+if test -f "$test_output_file"; then
+
+    # Write the results.json file based on the exit code of the command that was 
+    # just executed that tested the implementation file
+    failed=$(jq '.failed' $test_output_file)
+    if [ $failed -eq 0 ]; then
+        echo "${slug}: test passed"
+        jq -n '{version: 1, status: "pass"}' > ${results_file}
+    else
 
+        # OPTIONAL: Sanitize the output
+        # In some cases, the test output might be overly verbose, in which case stripping
+        # the unneeded information can be very helpful to the student
+        # sanitized_test_output=$(printf "${test_output}" | sed -n '/Test results:/,$p')
+
+        # OPTIONAL: Manually add colors to the output to help scanning the output for errors
+        # If the test output does not contain colors to help identify failing (or passing)
+        # tests, it can be helpful to manually add colors to the output
+        # colorized_test_output=$(echo "${test_output}" \
+        #      | GREP_COLOR='01;31' grep --color=always -E -e '^(ERROR:.*|.*failed)$|$' \
+        #      | GREP_COLOR='01;32' grep --color=always -E -e '^.*passed$|$')
```

Comment:
Feel free to remove this if you're not using it

Reply:
+1